lib/wraplibsnarkgadgets.cpp

```diff
@@ -0,0 +1,139 @@
+//https://gist.github.com/kobigurk/24c25e68219df87c348f1a78db51bb52
+
+#include <iostream>
+
+//#include "wraplibsnarkgadgets.cpp"
+
+/*#include "libsnark/gadgetlib1/gadget.hpp"
+#include "libsnark/gadgetlib1/protoboard.hpp"
+#include "libff/common/default_types/ec_pp.hpp"
+#include "libsnark/reductions/r1cs_to_qap/r1cs_to_qap.hpp"
+
+#include <libsnark/common/data_structures/merkle_tree.hpp>
+#include <libsnark/gadgetlib1/gadgets/basic_gadgets.hpp>
+#include <libsnark/gadgetlib1/gadgets/hashes/hash_io.hpp>
+#include <libsnark/gadgetlib1/gadgets/hashes/sha256/sha256_components.hpp>
+#include <libsnark/gadgetlib1/gadgets/hashes/sha256/sha256_gadget.hpp>*/
+#include "./sha256_ethereum.cpp"
+
+//r1cs to json
+#include <libsnark/zk_proof_systems/ppzksnark/r1cs_ppzksnark/r1cs_ppzksnark.hpp>
+
+
+using namespace libsnark;
+using namespace libff;
+
+using std::vector;
+
+
+template<typename FieldT>
```

Comment:
Do these functions need to be templated? 

I'm not sure how to easily call a templated C++ function from Rust (Rust only speaks C not C++) and if I'm reading the code right it's not required (there is a `typedef` for `FieldT` in [`sha256_ethereum`](https://github.com/JacobEberhardt/ZoKrates/pull/57/files#diff-ab7ad5652f9ddd2e8337327a30127bc0R22)).